src/main/java/com/code_intelligence/jazzer/runtime/TraceCmpHooks.java

```diff
@@ -213,6 +210,222 @@ private static void regionMatchesInternal(
+  @MethodHook(type = HookType.AFTER, targetClassName = "kotlin.text.StringsKt ",
+      targetMethod = "replaceBefore")
+  @MethodHook(type = HookType.AFTER, targetClassName = "kotlin.text.StringsKt ",
+      targetMethod = "replaceBefore$default")
+  @MethodHook(type = HookType.AFTER, targetClassName = "kotlin.text.StringsKt ",
+      targetMethod = "replaceBeforeLast")
+  @MethodHook(type = HookType.AFTER, targetClassName = "kotlin.text.StringsKt ",
+      targetMethod = "replaceBeforeLast$default")
+  @MethodHook(type = HookType.AFTER, targetClassName = "kotlin.text.StringsKt ",
+      targetMethod = "replaceFirst")
+  @MethodHook(type = HookType.AFTER, targetClassName = "kotlin.text.StringsKt ",
+      targetMethod = "replaceFirst$default")
+  public static void
+  replaceKt(
+      MethodHandle method, Object alwaysNull, Object[] arguments, int hookId, String returnValue) {
+    if (!(arguments[0] instanceof String)) {
+      return;
+    }
+    String original = (String) arguments[0];
+    if (!original.equals(returnValue)) {
+      return;
+    }
+
+    // We currently don't handle the overloads that take a regex as a second argument.
+    if (arguments[1] instanceof String || arguments[1] instanceof Character) {
+      TraceDataFlowNativeCallbacks.traceStrstr(original, arguments[1].toString(), hookId);
+    }
+  }
+
+  @MethodHook(type = HookType.AFTER, targetClassName = "kotlin.text.StringsKt ",
+      targetMethod = "regionMatches",
+      targetMethodDescriptor = "(Ljava/lang/String;ILjava/lang/String;IIZ)Z")
+  @MethodHook(type = HookType.AFTER, targetClassName = "kotlin.text.StringsKt ",
+      targetMethod = "regionMatches$default",
+      targetMethodDescriptor = "(Ljava/lang/String;ILjava/lang/String;IIZILjava/lang/Object;)Z")
+  public static void
+  regionMatchesKt(MethodHandle method, Object alwaysNull, Object[] arguments, int hookId,
+      Boolean doesRegionMatch) {
+    if (doesRegionMatch) {
+      return;
+    }
+
+    String thisString = arguments[0].toString();
+    int thisOffset = (int) arguments[1];
+    String other = arguments[2].toString();
+    int otherOffset = (int) arguments[3];
+    int length = (int) arguments[4];
+    regionMatchesInternal(thisString, thisOffset, other, otherOffset, length, hookId);
+  }
+
+  @MethodHook(type = HookType.AFTER, targetClassName = "kotlin.text.StringsKt ",
+      targetMethod = "indexOfAny")
+  @MethodHook(type = HookType.AFTER, targetClassName = "kotlin.text.StringsKt ",
+      targetMethod = "indexOfAny$default")
+  @MethodHook(type = HookType.AFTER, targetClassName = "kotlin.text.StringsKt ",
+      targetMethod = "lastIndexOfAny")
+  @MethodHook(type = HookType.AFTER, targetClassName = "kotlin.text.StringsKt ",
+      targetMethod = "lastIndexOfAny$default")
+  public static void
+  indexOfAnyKt(
+      MethodHandle method, Object alwaysNull, Object[] arguments, int hookId, Integer returnValue) {
+    if (arguments[0] instanceof CharSequence && returnValue == -1) {
```

Comment:
We don't check the type of `arguments[1]` here.

Reply:
There are two possible types for `arguments[1]`, and we check the type in `guideTowardContainmentOfFirstElement`.